tests/plugins/test_ddp_plugin.py

```diff
@@ -82,6 +86,105 @@ def on_fit_start(self, trainer, pl_module):
         trainer.fit(model)
 
 
+@mock.patch.dict(
+    os.environ,
+    {
+        "CUDA_VISIBLE_DEVICES": "0,1",
+        "SLURM_NTASKS": "2",
+        "SLURM_JOB_NAME": "SOME_NAME",
+        "SLURM_NODEID": "0",
+        "LOCAL_RANK": "0",
+        "SLURM_LOCALID": "0",
+    },
+)
+@mock.patch("torch.cuda.device_count", return_value=2)
+@pytest.mark.parametrize(
+    ["ddp_backend", "gpus", "num_processes"],
+    [("ddp_cpu", None, None), ("ddp", 2, 0), ("ddp2", 2, 0), ("ddp_spawn", 2, 0)],
+)
+@pytest.mark.skipif(platform.system() == "Windows", reason="Distributed sharded plugin is not supported on Windows")
+def test_ddp_choice_string_ddp_cpu(tmpdir, ddp_backend, gpus, num_processes):
+    class CB(Callback):
+        def on_fit_start(self, trainer, pl_module):
+            assert isinstance(trainer.accelerator_backend.ddp_plugin, DDPShardedPlugin)
+            raise SystemExit()
```

Comment:
```suggestion
            raise RuntimeError('finish test')
```
and later in 
```
    with pytest.raises(RuntimeError, match='finish test'):
        trainer.fit(model)
```

Reply:
but what you want here is just mock the call and check that it is called at least once, right?

Reply:
It's a little bit trickier than that, we want to ensure that the correct plugins get selected, but also terminate training before fit begins. This is because at fit time, DDP etc get initialized. Since we're mocking calls we need to terminate training before this

Reply:
Solution added :)